csharp/src/Microsoft.ML.OnnxRuntime/OnnxRuntime.shared.cs

```diff
@@ -50,16 +49,21 @@ public enum OrtLanguageProjection
     /// and other necessary things for OnnxRuntime to function. Create or access OrtEnv by calling
     /// the Instance() method. Call this method before doing anything else in your application.
     /// </summary>
-    public sealed class OrtEnv : SafeHandle
+    public sealed class OrtEnv : IDisposable
     {
-        private static readonly Lazy<OrtEnv> _instance = new Lazy<OrtEnv>(()=> new OrtEnv());
-        private static LogLevel envLogLevel = LogLevel.Warning;
+        private static Object _lock = new Object();
+        private static OrtEnv _instance = null;
+
+        private bool  _disposed = false;
+        private IntPtr handle = IntPtr.Zero;
+        private LogLevel envLogLevel = LogLevel.Warning;
```

Comment:
what about also making the log level configurable?

just an idea for another approach (my c# might be a bit rusty but hopefully the general idea comes across):
```c#
public sealed class OrtEnv : ... {
  // ...

  public class InitializationOptions {
    public LogLevel logLevel;
    public OrtThreadingOptions threadingOptions;
  }

  public static OrtEnv Instance(InitializationOptions initializationOptions) {
    // if initializationOptions is not null, throw an exception if instance is already created
    // create instance if needed and return it
  }

  public static OrtEnv Instance() { return Instance(null); }
}
```

we could only have one place where we need to do the double checked locking stuff. or maybe we could even keep using `Lazy<OrtEnv>` to avoid implementing it by hand.

Reply:
The log level is already configurable via a designated C API. I.e. it is a update. It is assumed that it may be updated more than once.

Reply:
We need to keep Instance() and only one of the params can be used (different API) . I think it would be misleading both for the client and for the implementation which one to use.

Reply:
I see, wasn't aware of the log level update.

but generally, was considering the extensibility in case in the future we need to add another way to create an env.

also, it would be nice if the complexity of synchronizing instance creation could be limited to one place. a creation helper function could call the different ORT APIs depending on the options.

Reply:
I agree. A single 'does everything' ctor with the complex code to do instance creation would be nice, with all other ctors calling that. 

Not allowing the log level to be set at construction time means the user cannot increase the default logger's severity to debug any issues during environment creation. That seems like a gap that would be good to close. I believe all the ORT C API functions to create the env allow the log level to be specified.

Reply:
There is an update API that allows you to change the log level, it is already exposed in C# API.